activity/sidecar.go

```diff
@@ -0,0 +1,222 @@
+package activity
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hootsuite/vault-ctrl-tool/aws"
+	"github.com/hootsuite/vault-ctrl-tool/cfg"
+	"github.com/hootsuite/vault-ctrl-tool/leases"
+	"github.com/hootsuite/vault-ctrl-tool/scrubber"
+	"github.com/hootsuite/vault-ctrl-tool/sshsigning"
+	"github.com/hootsuite/vault-ctrl-tool/util"
+	"github.com/hootsuite/vault-ctrl-tool/vaultclient"
+	jww "github.com/spf13/jwalterweatherman"
+	"golang.org/x/crypto/ssh"
+)
+
+func renewLeases(ctx context.Context, currentConfig cfg.Config, vaultClient vaultclient.VaultClient) {
+
+	threshold := time.Now().Add(util.Flags.RenewInterval).Add(util.Flags.SafetyThreshold)
+	jww.DEBUG.Printf("Will renew all credentials expiring before %v", threshold)
+
+	jww.DEBUG.Printf("Auth token: expires? %v at %v", leases.Current.AuthTokenLease.CanExpire, leases.Current.AuthTokenLease.ExpiresAt)
+	// Does our authentication token expire soon?
+	if leases.Current.AuthTokenLease.CanExpire {
+		if leases.Current.AuthTokenLease.ExpiresAt.Before(threshold) {
+
+			err := vaultClient.RenewSelf(ctx, util.Flags.RenewLeaseDuration)
+
+			if err != nil {
```

Comment:
If the next renewal interval is going to happen after the token expires, should we fatal here?